File: lib/progress.js
export function consoleProgress(event) {
  if (event.type === "log") {
    console.log(event.message);
  } else if (event.type === "item") {
    const idx = `[${event.index}/${event.total}]`;
    const name = (event.name || "").slice(0, 50);
    console.log(`  ${idx} ${name}… ${event.status || ""}`);
  } else if (event.type === "done") {
    if (event.message) console.log(event.message);
  } else if (event.type === "error") {
    console.error("erro:", event.message);
  }
}

File: python/progress.py
"""Progresso no console — equivalente a lib/progress.js."""


def console_progress(event):
    etype = event.get("type")
    if etype == "log":
        print(event.get("message", ""))
    elif etype == "item":
        idx = f"[{event.get('index')}/{event.get('total')}]"
        name = (event.get("name") or "")[:50]
        print(f"  {idx} {name}… {event.get('status', '')}")
    elif etype == "done":
        if event.get("message"):
            print(event["message"])
    elif etype == "error":
        print(f"erro: {event.get('message', '')}")
